lxd/db.go

```diff
@@ -57,31 +58,95 @@ CREATE TABLE images_properties (
     key VARCHAR(255) NOT NULL,
     value TEXT,
     FOREIGN KEY (image_id) REFERENCES images (id)
-);`
+);
+CREATE TABLE schema (
+    id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+    version INTEGER NOT NULL,
+    updated_at DATETIME NOT NULL,
+    UNIQUE (version)
+);
+INSERT INTO schema (version, updated_at) values (?, "now");`
 
-	_, err = db.Exec(stmt)
+	_, err = db.Exec(stmt, DB_CURRENT_VERSION)
+	return db, err
+}
+
+func getSchema(db *sql.DB) (int, error) {
+	rows, err := db.Query("SELECT max(version) FROM schema")
+	if err != nil {
+		return 0, nil
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var v int
+		rows.Scan(&v)
+		return v, nil
+	}
+	return 0, nil
+}
+
+func updateFromV0(db *sql.DB) error {
+	// v0..v1 adds schema table
+	stmt := `
+CREATE TABLE schema (
+    id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+    version INTEGER NOT NULL,
+    updated_at DATETIME NOT NULL,
+    UNIQUE (version)
+);
+INSERT INTO schema (version, updated_at) values (?, "now");`
+	_, err := db.Exec(stmt, DB_CURRENT_VERSION)
 	return err
 }
 
+func updateDb(db *sql.DB, prev_version int) error {
+	if prev_version < 0 || prev_version > DB_CURRENT_VERSION {
+		return fmt.Errorf("Bad database version: %d\n", prev_version)
+	}
+	if prev_version == DB_CURRENT_VERSION {
+		return nil
+	}
+	var err error
+	if prev_version == 0 {
+		err = updateFromV0(db)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
We're not doing a select inside the while statement.

We can streamline this as the list gets longer, but how to best do it will be most obvious when the need arises.